osu.Server.Spectator/Startup.cs

```diff
@@ -80,8 +86,21 @@ protected virtual void ConfigureAuthentication(IServiceCollection services)
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IHostApplicationLifetime lifetime)
         {
+            var shutdownManager = app.ApplicationServices.GetService<GracefulShutdownManager>();
+
+            Debug.Assert(shutdownManager != null);
+
+            // This is done here, but it happens too late in shutdown.
```

Comment:
What does "happens too late" mean here?

Reply:
SignalR is shut down first, making the whole thing fruitless.

Reply:
This also no longer exists.